### PR TITLE
[WIP] Validate IngressRoutes and update their status

### PR DESCRIPTION
**Pull request by alexbrand**
Updates #490 

Pending on #469

### CODE REVIEW

File: internal/dag/dag_test.go [hunk 1]
```diff
@@ -2939,11 +2954,11 @@ func TestDAGIngressRouteCycle(t *testing.T) {
 	})
 
 	want := make(map[hostport]*VirtualHost)
-	want[hostport{host: "example.com", port: 80}] = &VirtualHost{
-		Port:   80,
-		host:   "example.com",
-		routes: routemap(&Route{path: "/finance", object: ir2}),
-	}
+	// want[hostport{host: "example.com", port: 80}] = &VirtualHost{
```
Comment (davecheney) [inline internal/dag/dag_test.go]: please don't comment out code, just delete it.

Comment (alexbrand) [reply internal/dag/dag_test.go]: Whoops. Looks like a leftover debug artifact. Thanks!

**Comment by alexbrand** [comment]
Thanks @davecheney. Writing the status from within DAG.recompute also gave me a gut feeling that something was out of place, but I wasn't sure exactly what/where. I think your comments and suggestions crystallized the issue for me. I am going to close this PR and work on a new one.

**Comment by davecheney** [comment]
I think a good way forward would be to change recompute to return a slice of failed  validation types, I don’t know exactly goes inside that type, but some error value or string as a start as well as a reference to the original object. That should be enough of a toehold for a separate process to do status patches after the run.

**Pull request closed without merging**